src/sources/util/unix_stream.rs

```diff
@@ -11,12 +11,16 @@ use tokio_stream::wrappers::UnixListenerStream;
 use tokio_util::codec::FramedRead;
 use tracing::field;
 use tracing_futures::Instrument;
+use vector_core::ByteSizeOf;
 
 use crate::{
     async_read::VecAsyncReadExt,
     codecs,
     event::Event,
-    internal_events::{ConnectionOpen, OpenGauge, UnixSocketError, UnixSocketFileDeleteError},
+    internal_events::{
+        ConnectionOpen, OpenGauge, SocketEventsReceived, SocketMode, StreamClosedError,
+        UnixSocketError, UnixSocketFileDeleteError,
+    },
```

Comment:
Are we able to emit `BytesReceived` when using `unix_stream`? I don't see it here but might have missed it.

Reply:
when using `unix_stream` we are not able to do it, it is wrapped by the `FrameRead` that doesn't give a way to know how many bytes were being read.
We could put int just before the `Ok((mut events, byte_size)) => {` (line 84), but then it wouldn't comply with `before the creation of a Vector event` because the `events` are already vector events.

Reply:
after thinking about it, better late than never. I'll add it in the match 😉

Reply:
Hm, is this something we could emit at the `Deserializer` or `Decoder` level (e.g. [around here](https://github.com/vectordotdev/vector/blob/master/src/codecs/decoding/mod.rs#L259)) which has access to the original bytes? (some additional information seems like it'd have to be wired in and not sure how it'd affect other instrumentation and potentially performance). I think having `component_received_bytes_total` match `component_received_event_bytes_total` is not ideal. I'll defer to @bruceg and @jszwedko  on this.